nba_pagerank.py

```diff
@@ -1,32 +1,79 @@
 import csv
-
-
-class NbaPagerank(object):
+import numpy as np
+
+class PageRank(object):
+    def __init__(self, M, R):
+        if len(M) > 0:
+            assert len(M[0]) == len(M)
+        self.M = M
+        self.N = len(self.M) # M is a square matrix; N-by-N 2D numpy array.
+        self.R_0 = np.full((self.N, 1), 1 / float(self.N))
+        self.one_vector = np.ones((self.N, 1))
+
+    # see https://en.wikipedia.org/wiki/PageRank#Iterative
+    def run(self, iterations=100, d=0.85):
+        R = self.R_0
+        for i in range(iterations):
+            R = np.add(d * np.matmul(self.M, R), (1 - d) / float(self.N) * self.one_vector)
+        return R
+
+class NbaPageRank(PageRank):
     CSV_REGULAR_SEASON = 'data/2018_2019/regular_season.csv'
     CSV_PLAYOFFS = 'data/2018_2019/playoffs.csv'
 
+    def __init__(self, playoffs=False):
+        self.playoffs = playoffs
+        matches = self.matches_playoffs() if playoffs else self.matches_regular_season()
+        self.team_record_graph = self.build_graph(matches)
+        self.num_teams = len(self.team_record_graph)
```

Comment:
we can set the parent constructor's instance variable from `M`, so this can be an instance variable here.
```suggestion
        num_teams = len(self.team_record_graph)
```